### ISSUE TITLE
InstanceVariables check not working for multi-line blocks

### ISSUE THREAD

**Issue by andyw8** [sds/haml-lint#245]
The following generates an `InstanceVariables: Avoid using instance variables in partials views` warning, as expected:
```haml
= simple_form_for(@widget)
```
as does
```
= simple_form_for(@widget) { |f| f.input :name }
```
But this doesn't:
```haml
= simple_form_for(@widget) do |f|
  = f.input :name
```

Using haml-lint 0.24.0

I haven't yet tried on any other versions.

**Comment by michaelherold** [comment]
This is a weird one.

The `HamlLint::Tree::ScriptNode` class only has the first line (i.e. `= simple_form_for(@widget) do |f|`) as its script. This gets translated to the Ruby script ` simple_form_for(@widget) do |f|`. Since it's missing the `end` that goes with the block, the `::Parser::CurrentRuby` parse chokes on it as a syntax violation so we don't get the Ruby AST back from the parser.

I don't know how we can resolve this bug because I don't know how Haml handles the insertion of the implied `end`. Walking down the AST that we get at the Haml-Lint level, here are the two nodes:

```haml
= simple_form_for(@widget) do |f|
  = f.input :name
```

gets translated to:

```
[1] pry(#<HamlLint::Linter::InstanceVariables>)> node.script
=> " simple_form_for(@widget) do |f|"
[2] pry(#<HamlLint::Linter::InstanceVariables>)> node.children.map(&:script)
=> [" f.input :name"]
```

... which is equivalent to:

```ruby
simple_form_for(@widget) do |f|
  f.input :name
```

... which doesn't include the `end` for the block.

Without being able to send valid Ruby code to the Ruby parser, we won't be able to detect this problem.

Does anyone have any ideas on how we can handle this?

**Comment by sds** [comment]
Fixed in #364.
